doc/design/app-status.md

```diff
@@ -0,0 +1,93 @@
+### Status of my Application
+
+As a user, I expect to see the status of my application running across multiple clusters. When I go to my application list page I should be able to quickly see the "health" of each application in each cluster within a timely manner.
+
+## Problem to be Solved
+
+- Come up with a way for the CLI to have parity with the UI status request.
+- Determine how to get the status that will scale with larger clusters and multiple clusters.
+- Determine if there are unique statuses for helm or kustomize.
+
+## Outcome
+
+The application status is returned for all resources using https://github.com/kubernetes-sigs/cli-utils/tree/v0.26.1/pkg/kstatus. The UI is currently using this to generate the status shown in this graph: 
+![Screen Shot 2021-12-09 at 1.32.11 PM.png](https://images.zenhubusercontent.com/60f5a90306cf76325d5accdc/be24ca61-a0dd-47d6-a847-240b38f422af)
+
+*_Only the UI is using this currently._
+
+The CLI currently looks like:
+```
+NAME                       READY   MESSAGE                                                         REVISION                                        SUSPENDED 
+gitrepository/pod-info     True    Fetched revision: b7b8ec639555c425ff8a5d61174d974f733ab6b0      b7b8ec639555c425ff8a5d61174d974f733ab6b0        False    
+
+NAME                          READY   MESSAGE                                 REVISION        SUSPENDED 
+kustomization/pod-info        True    Release reconciliation succeeded        2.8.2           False    
+```
+
+The UI will remain the same while the CLI will need to be updated like so to match: 
+```
+***Application details***:
+  Branch:           main
+  config_url:       ssh://git@github.com/sympatheticmoose/weave-gitops-multicluster-demo.git
+  deployment_type:  kustomize
+  Path:             ./
+  source_type:      git
+  URL:              ssh://git@github.com/sympatheticmoose/podinfo-deploy.git
+
+***Reconciliation details***:
+Last successful reconciliation: 2022-01-06 10:43:47 +0000 GMT
+
+NAME                 	READY	MESSAGE                                                        	REVISION                                     	SUSPENDED 
+gitrepository/podinfo	True 	Fetched revision: main/1e644628deb645c454f4db167e1041e95e2d0ba3	main/1e644628deb645c454f4db167e1041e95e2d0ba3	False    	
+
+NAME                 	READY	MESSAGE                                                        	REVISION                                     	SUSPENDED 
+kustomization/podinfo	True 	Applied revision: main/1e644628deb645c454f4db167e1041e95e2d0ba3	main/1e644628deb645c454f4db167e1041e95e2d0ba3	False  
+
+***Deployment details***:
+KIND              NAME                     STATUS
+Application       josh-pod-info            ready
+Deployment        backend                  ready
+ReplicaSet        backend-6b944d8b         ready
+Pod               backend-6b944d8b-tzxnl   ready
+Deployment        frontend                 ready
+Kustomization     podinfo-deploy           ready
+```
```

Comment:
There is a discussion about how much info to show on here as well as how to show the hierarchy. Any ideas would be helpful.